### PR TITLE
Bump submodule with filesystem interceptor and test fixes

### PR DESCRIPTION
**Pull request by davidtrevelyan**

### CODE REVIEW

**Comment by cjappl** [comment]
I'm wondering if we got sandboxed here somehow in the ubuntu machine. It appears that the permissions we got on that test were:

```
Value of: st.st_mode & 0777
Expected: is equal to 420
  Actual: 384
```

Translating from hex: 
we got            384 (0x600)   (User read write, nothing else)                    `rw-------`
We expected 420 (0x0644) (User read write, group read, others read)  `rw-r--r--`

This possibly means that we are just not allowed on this VM to create files with permissions for others and groups. Or it means I messed up somewhere in the implementation, but considering this is green on the big review, and green locally on ubuntu for me makes me more confident that isn't the case.


The only thing I can think of doing while keeping this test mostly intact is something like this:

```cpp
TEST(TestRadsanInterceptors, openCreatesFileWithProperMode) {


  for (int mode : {S_IWUSR, S_IRUSR}) {     // LOOK HERE!! WE TEST ONLY ON THE USER PERM MODES


    const int fd = open(temporary_file_path(), O_CREAT | O_WRONLY, mode);
    ASSERT_THAT(fd, Ne(-1)) << "Could not create file, mode: " << mode;  
    close(fd);

    struct stat st;
    ASSERT_THAT(stat(temporary_file_path(), &st), Eq(0));

    // Mask st_mode to get permission bits only
    ASSERT_THAT(st.st_mode & 0777, Eq(mode));

    std::remove(temporary_file_path());
  }
}
```

Basically just test both of the user level permissions, and ensure they are passed through. I believe this test to still be valid as it makes sure the third argument isn't dropped on the floor but does not use the higher level masks that may be sandboxed on this machine.

Thoughts?

**Comment by cjappl** [comment]
Actually, on second look I think we may just be stomping on the same file multiple times and having an issue when our test run in multi-processing/threading. Check out this PR for more https://github.com/realtime-sanitizer/llvm-project/pull/25.

I was noticing that when trying to repro this issue, I would fail once every 10 times with some file permissions of another test. Putting `std::remove(..)` as the top line of this test fixed the issue, making me think we were just getting some cross talk between tests.

**Review by cjappl — APPROVED** [review]
Green tests, approved ✅
